src/styles/molecules/_search-form.scss

```diff
@@ -15,10 +15,15 @@
 }
 
 @mixin dc-search-form__input {
+    max-height: 36px;
     margin: 0;
     padding-right: 30px;
     -moz-appearance: none;
     -webkit-appearance: none;
+
+    &[type="date"]::-webkit-clear-button {
```

Comment:
@LeoAref http://stackoverflow.com/questions/19953333/remove-clear-button-on-date-input-ios-safari

Check this apparently there is also something for targeting IE....